module-new/src/components/characterLimit/characterLimit.component.tsx

```diff
@@ -0,0 +1,68 @@
+import * as React from 'react';
+
+import { IBindingProps, useBindingState } from '../../form';
+import { ArmstrongFCExtensions, ArmstrongFCProps, ArmstrongFCReturn, NullOrUndefined } from '../../types';
+import { concat } from '../../utils/classNames';
+import { useArmstrongConfig } from '../config';
+
+import './characterLimit.theme.css';
+
+export interface ICharacterLimitProps<TBind extends NullOrUndefined<string>>
+  extends Omit<React.RefAttributes<HTMLDivElement>, 'ref'> {
+  /**  prop for binding to an Armstrong form binder (see forms documentation) */
+  bind: IBindingProps<TBind>;
+
+  /** the current value of the string to count */
+  value?: TBind;
+
+  /** the character limit for the bound input */
+  limit: number;
+
+  /** the limit should be enforced by the bind in this component - by default you will have to handle this yourself */
+  shouldEnforce?: boolean;
+
+  /** CSS className property */
+  className?: string;
+
+  /** the icon to use for the validation errors */
+  validationErrorIcon?: JSX.Element;
+}
+
+/** Render a character limit from a bound value, showing as an error if the user  */
+export const CharacterLimit = React.forwardRef<HTMLDivElement, ICharacterLimitProps<NullOrUndefined<string>>>(
+  ({ bind, limit, shouldEnforce, value, className, validationErrorIcon, ...nativeProps }, ref) => {
+    const globals = useArmstrongConfig({
+      validationErrorIcon,
+    });
+
+    const [boundValue, setBoundValue] = useBindingState(bind, { value });
+
+    const exceeded = boundValue && boundValue.length > limit;
```

Comment:
Should this be memoized? It's used as a dependency in the next effect so won't this cause the effect to trigger every render unless it's in a memo?

Reply:
It won't no, because the resulting value is a primitive, and primitive values when used as dependencies are evaluated by value not by reference.

So if `exceeded` is `false` on render 1 and then `false` again on render 2, then the effect won't fire on render 2 because `false === false`. If I was composing a new array or object then that would be different because the `===` checks the reference not the value.